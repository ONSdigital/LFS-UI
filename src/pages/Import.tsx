import React, {Component} from "react";
import {ONSUpload} from "../components/ONS_DesignSystem/ONSUpload";
import {ONSButton} from "../components/ONS_DesignSystem/ONSButton";
import {postImportFile} from "../utilities/http";
import {ONSPanel} from "../components/ONS_DesignSystem/ONSPanel";
import {FileUploadProgress} from "./FileUploadProgress";
import {isDevEnv, toUpperCaseFirstChar} from "../utilities/Common_Functions";
import DocumentTitle from "react-document-title";
import {ONSDateInput} from "../components/ONS_DesignSystem/ONSDateInput";
import {ReportExport} from "../components/ReportExport";

interface Props {
    match: any
}

interface State {
    uploadFile: any
    uploading: boolean
    validFromDate: Date | null
    importName: string
    importHidden: boolean
    uploadProgressHidden: boolean
    validFromDateHidden: boolean
    reportExportHidden: boolean
    fileType: string
    uploadLink: string
    //check to see if functionality is built and whether to send the request
    built: boolean
    hasImportReport: boolean
    fileName: string
    panel: Panel
}

interface Panel {
    label: string,
    visible: boolean
    status: string
}

export class Import extends Component <Props, State> {
    displayName = Import.name;

    constructor(props: Props) {
        super(props);

        let importName = (props.match.params.file !== undefined ? props.match.params.file : "");
        this.state = {
            uploadFile: "",
            uploading: false,
            validFromDate: null,
            importName: importName,
            importHidden: true,
            uploadProgressHidden: true,
            validFromDateHidden: true,
            reportExportHidden: true,
            fileType: "",
            uploadLink: "",
            built: false,
            fileName: "string",
            hasImportReport: false,
            panel: {
                label: '',
                visible: false,
                status: 'info'
            }
        };
        this.setPanel.bind(this);
        this.setFileUploading.bind(this);
        this.handleFileChange = this.handleFileChange.bind(this);
    }

    componentDidMount(): void {
        console.log(this.state.importName);
        if (this.state.importName.length > 0) {
            this.fileType(this.state.importName);
            this.setState({
                importHidden: false,
                uploadProgressHidden: true,
            });
        }
    }

    upload = () => {
        console.log('Uploading File');
        this.hidePanel();
        this.setState({
            uploading: true,
        });
        // TODO: Send correct data for each file type
        if (this.state.built) {
            let uploadLink = this.state.uploadLink;
            if (this.state.validFromDate !== null) {
                uploadLink = uploadLink + "/" + this.state.validFromDate.toISOString()
            }
            postImportFile(this.state.uploadFile, uploadLink, this.state.fileName)
                .then(response => {
                    (isDevEnv && console.log(response));
                    if (response.status === 'ERROR') {
                        this.setPanel(response.errorMessage.toString(), 'error');
                        this.setState({importHidden: true, uploadProgressHidden: true})
                    } else {
                        if (response.status === "OK") {
                            this.setPanel(toUpperCaseFirstChar(this.state.importName) + ': File Uploaded Successfully', 'success');
                        } else if (response.status === "INFO" && response.filename === "design_weights") {
                            this.setPanel(toUpperCaseFirstChar(this.state.importName) + ": File Uploaded Successfully, " + response.message, 'info');
                        }
                    }
                    this.setState({
                        uploading: false,
                    });
                })
                .catch(err => {
                    console.log(err);
                    this.setPanel(err.toString(), 'error',);
                    this.setState({
                        uploading: false,
                        uploadProgressHidden: false
                    });
                });
        } else {
            this.setPanel(this.state.importName + ': Import Not Implemented Yet', 'error');
            this.setState({
                uploading: false,
            });
        }
    };

    hidePanel = () => {
        this.setState({
            panel: {
                label: "",
                visible: false,
                status: "info"
            }
        });
    };

    setPanel = (message: string, status: string, visible: boolean = true) => {
        this.setState({
            panel: {
                label: message,
                visible: visible,
                status: status
            }
        });
    };

    handleFileChange = (selectorFiles: FileList | null) => {
        console.log(selectorFiles);
        this.setState({uploadFile: selectorFiles});
    };

    handleDateChange = (date: Date) => {
        this.setState({validFromDate: date});
    };

    setFileUploading = (bool: boolean) => {
        this.setState({importHidden: bool});
        if (this.state.hasImportReport) {
            this.setState({reportExportHidden: false})
        }
    };

    fileType = (file: string) => {
        switch (file) {
            case "Geographical Classifications":
                this.setState({
                    fileType: '.csv',
                    built: true,
                    fileName: "address",
                    uploadLink: 'address',
                    validFromDateHidden: true,
                    hasImportReport: false
                });
                break;
            case "Bulk Amendments":
                this.setState({});
                break;
            case "Design Weights":
                this.setState({
                    fileType: '.csv',
                    built: true,
                    fileName: "design_weights",
                    uploadLink: 'design/weights',
                    validFromDateHidden: true,
                    hasImportReport: false
                });
                break;
            case "Population Estimates":
                this.setState({
                    fileType: '.xlsx',
                    built: true,
                    fileName: "population",
                    uploadLink: 'population',
                    validFromDateHidden: true,
                    hasImportReport: true
                });
                break;
            case "Value Labels":
                this.setState({
                    fileType: '.csv',
                    built: true,
                    fileName: "value_labels",
                    uploadLink: 'value/labels',
                    validFromDateHidden: false,
                    hasImportReport: false
                });
                break;
            case "Variable Definitions":
                this.setState({
                    fileType: '.csv',
                    built: true,
                    fileName: "variable_definitions",
                    uploadLink: 'variable/definitions',
                    validFromDateHidden: false,
                    hasImportReport: false
                });
                break;
        }
    };

    render() {
        return (
            <DocumentTitle title={'LFS Import ' + this.state.importName}>
                <div className="container">
                    <form>
                        <ONSPanel status={this.state.panel.status} label={this.state.panel.label}
                                  hidden={!this.state.panel.visible}>
                            <p>{this.state.panel.label}</p>
                        </ONSPanel>
                        <br/>
                        <div hidden={this.state.importHidden}>
                            <div hidden={this.state.validFromDateHidden}>
                                <ONSDateInput label="Select Valid From Date" onChange={this.handleDateChange}
                                              date={this.state.validFromDate}/>
                                <br/><br/>
                            </div>
                            <ONSUpload label={"Import " + toUpperCaseFirstChar(this.state.importName)}
                                       description={"Only " + this.state.fileType + " accepted"} fileName={"Upload 1"}
                                       fileID={"U1"}
                                       accept={this.state.fileType}
                                       onChange={(e) => this.handleFileChange(e.target.files)}/>
                            <ONSButton label={"Submit"}
                                       field={true}
                                       onClick={this.upload}
                                       primary={true}
                                       loading={this.state.uploading}/>
                            <ONSButton label={'Return to Manage Batch'}
                                       primary={false}
                                       field={true}
                                       onClick={() => window.history.back()}/>
                            <ReportExport hidden={this.state.reportExportHidden} setPanel={this.setPanel}
                                          importName={this.state.uploadLink}/>
                        </div>
                    </form>
                    <FileUploadProgress importName={this.state.importName}
                                        fileName={this.state.fileName}
                                        hidden={this.state.uploadProgressHidden}
                                        fileUploading={this.setFileUploading}
                                        setPanel={this.setPanel}/>
                </div>
            </
                DocumentTitle>
        )
    }
}