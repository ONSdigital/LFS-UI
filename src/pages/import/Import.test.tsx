import React from "react"

import {Import} from "./Import"

import { default as flushPromises } from "../../tests/util/flushPromises"
import {render, fireEvent, screen, cleanup} from '@testing-library/react'

import { BrowserRouter } from "react-router-dom"
import { act } from "react-dom/test-utils";

import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import fetch from '../../tests/setup/__mocks__/fetch.js';

import '@testing-library/jest-dom';
import { getMonthandYear } from "../../tests/util/getMonthandYear"

// @ts-ignore 
global.fetch = fetch;
// @ts-ignore 
global.URL.createObjectURL = jest.fn();

// @ts-ignore 
if (global.document)
  document.createRange = () => ({
    setStart: () => {},
    setEnd: () => {},
    // @ts-ignore 
    commonAncestorContainer: {
      nodeName: "BODY",
      ownerDocument: document,
    },
})

const designWeightsMatch = {
    path: "/import/:file?",
    url: "/import/APS Design Weights",
    isExact: true,
    params: {file: "APS Design Weights"}
}

const bulkAmendmentsMatch = {
    path: "/import/:file?",
    url: "/import/Bulk Amendments",
    isExact: true,
    params: {file: "Bulk Amendments"}
}

const geogClassMatch = {
    path: "/import/:file?",
    url: "/import/Geographical Calculations",
    isExact: true,
    params: {file: "Geographical Calculations"}
}

const populationEstimatesMatch = {
    path: "/import/:file?",
    url: "/import/Population Estimates",
    isExact: true,
    params: {file: "Population Estimates"}
}

const valueLabelsMatch = {
    path: "/import/:file?",
    url: "/import/Value Labels",
    isExact: true,
    params: {file: "Value Labels"}
}

const variableDefinitionsMatch = {
    path: "/import/:file?",
    url: "/import/Variable Definitions",
    isExact: true,
    params: {file: "Variable Definitions"}
}

const APSHouseholdMatch = {
    path: "/import/:file?",
    url: "/import/APS Household Variable Specification",
    isExact: true,
    params: {file: "APS Household Variable Specification"}
}

const APSPersonMatch = {
    path: "/import/:file?",
    url: "/import/APS Person Variable Specification",
    isExact: true,
    params: {file: "APS Person Variable Specification"}
}

const eurostatMatch = {
    path: "/import/:file?",
    url: "/import/Eurostat Variable Specification",
    isExact: true,
    params: {file: "Eurostat Variable Specification"}
}

const LFSHouseholdMatch = {
    path: "/import/:file?",
    url: "/import/LFS Household Variable Specification",
    isExact: true,
    params: {file: "LFS Household Variable Specification"}
}

const LFSPersonMatch = {
    path: "/import/:file?",
    url: "/import/LFS Person Variable Specification",
    isExact: true,
    params: {file: "LFS Person Variable Specification"}
}


function Props(input: any) {
    return {
        match: input
    }
}

function wrapper(render: any, props: any){
        return render(
            <BrowserRouter>
                <Import {...props}/>
            </BrowserRouter>
        )
}

let renderAndImport = (match: any, render: any, fileName?: string) => {
    wrapper(render, Props(match))

    const inputEl = screen.getByLabelText(/import/i);

    let file;
    if (fileName) file = new File(["(⌐□_□)"], fileName + ".csv", {type: "csv"});
    else file = new File(["(⌐□_□)"], match.params.file + ".csv", {type: "csv"});

    Object.defineProperty(inputEl, "files", {
        value: [file]
    });
    
    fireEvent.change(inputEl);
}

Enzyme.configure({adapter: new Adapter()});

describe("Import page snapshot rendering", () => {
    // Here we are testing if each import renders and matches the snapshot
    afterEach(cleanup);
    
    // Imports visible on import overview
    it("renders the APS design weights page correctly", () => {
        expect(wrapper(render, Props(designWeightsMatch))).toMatchSnapshot()
    })

    it("renders the bulk amendments page correctly", () => {
        expect(wrapper(render, Props(bulkAmendmentsMatch))).toMatchSnapshot()
    })

    it("renders the geographical classifications page correctly", () => {
        expect(wrapper(render, Props(geogClassMatch))).toMatchSnapshot()
    })

    it("renders the population estimates page correctly", () => {
        expect(wrapper(render, Props(populationEstimatesMatch))).toMatchSnapshot()
    })

    it("renders the value labels page correctly", () => {
        expect(wrapper(render, Props(valueLabelsMatch))).toMatchSnapshot()
    })
    
    it("renders the variable definitions page correctly", () => {
        expect(wrapper(render, Props(variableDefinitionsMatch))).toMatchSnapshot()
    })

    //Imports from output spec
    it("renders the APS household variable specification page correctly", () => {
        expect(wrapper(render, Props(APSHouseholdMatch))).toMatchSnapshot()
    })

    it("renders the APS person variable specification page correctly", () => {
        expect(wrapper(render, Props(APSPersonMatch))).toMatchSnapshot()
    })

    it("renders the eurostat variable specification page correctly", () => {
        expect(wrapper(render, Props(eurostatMatch))).toMatchSnapshot()
    })

    it("renders the LFS household specification page correctly", () => {
        expect(wrapper(render, Props(LFSHouseholdMatch))).toMatchSnapshot()
    })

    it("renders the LFS person specification page correctly", () => {
        expect(wrapper(render, Props(LFSPersonMatch))).toMatchSnapshot()
    })
})

describe("Import page functionality", () => {
    // Here we are testing the page functionality covering all the states the page could be in
    afterEach(cleanup);

    test("breadcrumb points to the correct url", async () => {
        //testing an output file as they have two breadcrumbs
        let breadWrapper = wrapper(mount, Props(APSPersonMatch))
        
        //Home breadcrumb links to Home
        expect(breadWrapper.find("ONSBreadcrumbs").getElement().props.List[0].link).toEqual("import/overview")

        //manage batch breadcrumb links to the correct manage batch page
        expect(breadWrapper.find("ONSBreadcrumbs").getElement().props.List[1].link).toEqual('import/output')
    })

    it('displays "no file selected" panel when the upload doesnt contain a file', async () => {
        const { getByText }  = wrapper(render, Props(designWeightsMatch))

        fireEvent.click(getByText(/submit/i))

        expect(getByText(/No File Selected/i)).toBeTruthy();
    });

    it('shows the correct panels and correct date for the date input', async () => {
        renderAndImport(variableDefinitionsMatch, render)
        
        fireEvent.click(screen.getByText(/submit/i))

        //no date has been selected when submitting so 'no date selected panel' shows
        expect(screen.getByText(/no date selected/i))
        
        fireEvent.click(screen.getByTestId(/text-input/i))
        
        fireEvent.click(screen.getByText(String(17)))
        
        //date has been selected and is displayed in the date input
        expect(screen.getByTestId(/text-input/i)).toHaveAttribute("value", getMonthandYear("/", 17))

        fireEvent.click(screen.getByText(/submit/i))

        await act(async () => {
            await flushPromises();
        });

        // Correct file is uploaded and shows the correct panel
        expect(screen.getByText(/Variable Definitions: File Uploaded Successfully/i))
    })

    // TODO
    // test('the dropdowns on the output spec imports function correctly', async () => {
    //     renderAndImport(APSHouseholdMatch, render)

    //     fireEvent.click(screen.getByText(/submit/i))

    //     // File submitted with no year
    //     expect(screen.getAllByText(/please select a year/i))
    //     // console.log(screen.getByLabelText(/year/i))
    //     console.log(screen.getByTestId(/option-year/i))

    //     fireEvent.click(screen.getByTestId(/option-year/i))
    //     // console.log(screen.getByLabelText(/year/i))
    //     console.log(screen.getByTestId(/option-year/i))
    //     fireEvent.click(screen.getByText("2020"))
    //             expect(screen.getByLabelText(/yearrrr/i)).toHaveAttribute("value", "2020")

    // })
})

describe("Importing files and handling", () => {
    // Here we are testing the imports which are distinctive, they have a websocket or modal etc
    afterEach(cleanup);

    it('selects a design weights file and uploads, returning an OK response and displaying a success panel', async () => {
        renderAndImport(designWeightsMatch, render)
        
        fireEvent.click(screen.getByText(/submit/i))

        await act(async () => {
            await flushPromises();
        });
        
        expect(screen.getByText(/APS Design Weights: File Uploaded Successfully/i)).toBeTruthy();
    });

    it('selects a population estimates file and uploads, returning an OK response, displaying a success panel and a download button', async () => {
        renderAndImport(populationEstimatesMatch, render)

        fireEvent.click(screen.getByText(/submit/i))

        await act(async () => {
            await flushPromises();
        });

        expect(screen.getByText(/Population Estimates: File Uploaded Successfully/i)).toBeTruthy();

        fireEvent.click(screen.getByText(/download report/i))
        


        await act(async () => {
            await flushPromises();
        });

        // Expect the panel to show the report has been exported
        expect(screen.getByText(/Report Exported/i)).toBeTruthy();

    })

    it('selects a bulk ammendments file and validates, returning an OK response, displaying a modal, then the file is accepted and imported', async () => {
        renderAndImport(bulkAmendmentsMatch, render)

        fireEvent.click(screen.getByText(/submit/i))

        await act(async () => {
            await flushPromises();
        });

        // Bulk amendment file validates and responds with success
        expect(screen.getByText(/validation completed with no errors/i)).toBeTruthy();

        fireEvent.click(screen.getByTestId(/accept-button/i))
        
        await act(async () => {
            await flushPromises();
        });

        // Bulk amendment file imports and responds with success
        expect(screen.getAllByText(/Bulk Amendments: File Uploaded Successfully/i)).toBeTruthy();

        // Close the modal
        fireEvent.click(screen.getByText(/close/i))

        // Return to import overview
        fireEvent.click(screen.getByText(/return to import overview/i))

    })

    it('selects a bulk ammendments file and validates, returning an unmatched response, displaying a modal, it is then rejected', async () => {
        renderAndImport(bulkAmendmentsMatch, render, "Bulk Amendments Reject")

        fireEvent.click(screen.getByText(/submit/i))

        await act(async () => {
            await flushPromises();
        });

        // Bulk amendment file validates and responds with success
        expect(screen.getAllByText(/Unmatched items in Bulk Amendments file/i)).toBeTruthy();

        fireEvent.click(screen.getByText(/reject/i))

        // Modal is closed when reject button is clicked
        expect(screen.queryByText(/reject/i)).toBeNull()
    })
})