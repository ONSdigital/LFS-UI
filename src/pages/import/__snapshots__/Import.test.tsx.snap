// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import page functionality renders the page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <nav
          aria-label="Breadcrumb"
          class="breadcrumb"
          data-testid="breadcrumbs"
        >
          <ol
            class="breadcrumb__items"
          >
            <li
              class="breadcrumb__item u-fs-s"
              id="breadcrumb-1"
            >
              <a
                class="breadcrumb__link"
                href="/import/overview"
              >
                Import Overview
              </a>
            </li>
            <li
              class="breadcrumb__item u-fs-s breadcrumb__item--current"
              data-testid="breadcrumb-current"
              id="breadcrumb-2"
            >
              APS Design Weights
            </li>
          </ol>
        </nav>
        <h3>
          Import 
          APS Design Weights
        </h3>
        <br />
        <form>
          <div>
            <div
              hidden=""
            >
              <p
                class="field"
              >
                <label
                  class="label"
                >
                  Year
                </label>
                <select
                  class="input input--select"
                  id="year"
                  name="select"
                >
                  <option
                    disabled=""
                    selected=""
                    value=""
                  >
                    Select an option
                  </option>
                  <option
                    value="2020"
                  >
                    2020
                  </option>
                  <option
                    value="2019"
                  >
                    2019
                  </option>
                  <option
                    value="2018"
                  >
                    2018
                  </option>
                  <option
                    value="2017"
                  >
                    2017
                  </option>
                  <option
                    value="2016"
                  >
                    2016
                  </option>
                  <option
                    value="2015"
                  >
                    2015
                  </option>
                  <option
                    value="2014"
                  >
                    2014
                  </option>
                  <option
                    value="2013"
                  >
                    2013
                  </option>
                  <option
                    value="2012"
                  >
                    2012
                  </option>
                  <option
                    value="2011"
                  >
                    2011
                  </option>
                </select>
              </p>
              <p
                class="field"
              >
                <label
                  class="label"
                >
                  Quarter
                </label>
                <select
                  class="input input--select"
                  id="quarter"
                  name="select"
                >
                  <option
                    disabled=""
                    selected=""
                    value=""
                  >
                    Select an option
                  </option>
                  <option
                    value="Q1"
                  >
                    Q1
                  </option>
                  <option
                    value="Q2"
                  >
                    Q2
                  </option>
                  <option
                    value="Q3"
                  >
                    Q3
                  </option>
                  <option
                    value="Q4"
                  >
                    Q4
                  </option>
                </select>
              </p>
              <br />
            </div>
            <div
              class="field"
            >
              <p
                class="field"
              >
                <label
                  class="label"
                  for="U1"
                >
                  Import File
                  <br />
                  <span
                    class="label__description"
                  >
                    Only .csv accepted
                  </span>
                </label>
                <input
                  accept=".csv"
                  class="input input--text input-type__input input--upload"
                  id="U1"
                  name="Upload 1"
                  style="position: static;"
                  type="file"
                />
              </p>
            </div>
            <button
              class="btn field "
              data-testid="button"
              type="button"
            >
              <span
                class="btn__inner"
              >
                Submit
              </span>
            </button>
            <button
              class="btn field btn--secondary "
              data-testid="button"
              type="button"
            >
              <span
                class="btn__inner"
              >
                Return to Import Overview
              </span>
            </button>
            <div
              hidden=""
            >
              <br />
              <button
                class="btn   btn--excel btn--secondary"
                data-testid="button"
                type="button"
              >
                <span
                  class="btn__inner"
                >
                  Download Report
                </span>
              </button>
              <br />
              <br />
            </div>
          </div>
        </form>
        <br />
        <div
          hidden=""
        >
          <h4>
            Import Progress
          </h4>
          <div
            style="width: 60%;"
          >
            <table
              class="table table--sortable"
              data-aria-asc="ascending"
              data-aria-desc="descending"
              data-aria-sort="Sort by"
              id="basic-table"
            >
              <thead
                class="table__head"
              >
                <tr
                  class="table__row"
                >
                  <th
                    class="table__header "
                    scope="col"
                  >
                    File Name 
                  </th>
                  <th
                    class="table__header "
                    scope="col"
                  >
                    Step
                  </th>
                  <th
                    class="table__header "
                    scope="col"
                  >
                    Status
                  </th>
                </tr>
              </thead>
              <tbody
                class="table__body "
              >
                <tr
                  class="table__row nonSelectableTableRow"
                  data-testid="table-row"
                  tabindex="0"
                  title="Table Row"
                >
                  <td
                    class="table__cell "
                  />
                  <td
                    class="table__cell "
                  >
                    Import
                  </td>
                  <td
                    class="table__cell "
                  >
                    <span
                      class="status status--info"
                    />
                  </td>
                </tr>
              </tbody>
            </table>
            <div
              class="field"
            >
              <div
                class="progress-bar"
              >
                <div
                  class="progress-bar-filler"
                  style="width: 0%; background: rgb(136, 136, 136);"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <nav
        aria-label="Breadcrumb"
        class="breadcrumb"
        data-testid="breadcrumbs"
      >
        <ol
          class="breadcrumb__items"
        >
          <li
            class="breadcrumb__item u-fs-s"
            id="breadcrumb-1"
          >
            <a
              class="breadcrumb__link"
              href="/import/overview"
            >
              Import Overview
            </a>
          </li>
          <li
            class="breadcrumb__item u-fs-s breadcrumb__item--current"
            data-testid="breadcrumb-current"
            id="breadcrumb-2"
          >
            APS Design Weights
          </li>
        </ol>
      </nav>
      <h3>
        Import 
        APS Design Weights
      </h3>
      <br />
      <form>
        <div>
          <div
            hidden=""
          >
            <p
              class="field"
            >
              <label
                class="label"
              >
                Year
              </label>
              <select
                class="input input--select"
                id="year"
                name="select"
              >
                <option
                  disabled=""
                  selected=""
                  value=""
                >
                  Select an option
                </option>
                <option
                  value="2020"
                >
                  2020
                </option>
                <option
                  value="2019"
                >
                  2019
                </option>
                <option
                  value="2018"
                >
                  2018
                </option>
                <option
                  value="2017"
                >
                  2017
                </option>
                <option
                  value="2016"
                >
                  2016
                </option>
                <option
                  value="2015"
                >
                  2015
                </option>
                <option
                  value="2014"
                >
                  2014
                </option>
                <option
                  value="2013"
                >
                  2013
                </option>
                <option
                  value="2012"
                >
                  2012
                </option>
                <option
                  value="2011"
                >
                  2011
                </option>
              </select>
            </p>
            <p
              class="field"
            >
              <label
                class="label"
              >
                Quarter
              </label>
              <select
                class="input input--select"
                id="quarter"
                name="select"
              >
                <option
                  disabled=""
                  selected=""
                  value=""
                >
                  Select an option
                </option>
                <option
                  value="Q1"
                >
                  Q1
                </option>
                <option
                  value="Q2"
                >
                  Q2
                </option>
                <option
                  value="Q3"
                >
                  Q3
                </option>
                <option
                  value="Q4"
                >
                  Q4
                </option>
              </select>
            </p>
            <br />
          </div>
          <div
            class="field"
          >
            <p
              class="field"
            >
              <label
                class="label"
                for="U1"
              >
                Import File
                <br />
                <span
                  class="label__description"
                >
                  Only .csv accepted
                </span>
              </label>
              <input
                accept=".csv"
                class="input input--text input-type__input input--upload"
                id="U1"
                name="Upload 1"
                style="position: static;"
                type="file"
              />
            </p>
          </div>
          <button
            class="btn field "
            data-testid="button"
            type="button"
          >
            <span
              class="btn__inner"
            >
              Submit
            </span>
          </button>
          <button
            class="btn field btn--secondary "
            data-testid="button"
            type="button"
          >
            <span
              class="btn__inner"
            >
              Return to Import Overview
            </span>
          </button>
          <div
            hidden=""
          >
            <br />
            <button
              class="btn   btn--excel btn--secondary"
              data-testid="button"
              type="button"
            >
              <span
                class="btn__inner"
              >
                Download Report
              </span>
            </button>
            <br />
            <br />
          </div>
        </div>
      </form>
      <br />
      <div
        hidden=""
      >
        <h4>
          Import Progress
        </h4>
        <div
          style="width: 60%;"
        >
          <table
            class="table table--sortable"
            data-aria-asc="ascending"
            data-aria-desc="descending"
            data-aria-sort="Sort by"
            id="basic-table"
          >
            <thead
              class="table__head"
            >
              <tr
                class="table__row"
              >
                <th
                  class="table__header "
                  scope="col"
                >
                  File Name 
                </th>
                <th
                  class="table__header "
                  scope="col"
                >
                  Step
                </th>
                <th
                  class="table__header "
                  scope="col"
                >
                  Status
                </th>
              </tr>
            </thead>
            <tbody
              class="table__body "
            >
              <tr
                class="table__row nonSelectableTableRow"
                data-testid="table-row"
                tabindex="0"
                title="Table Row"
              >
                <td
                  class="table__cell "
                />
                <td
                  class="table__cell "
                >
                  Import
                </td>
                <td
                  class="table__cell "
                >
                  <span
                    class="status status--info"
                  />
                </td>
              </tr>
            </tbody>
          </table>
          <div
            class="field"
          >
            <div
              class="progress-bar"
            >
              <div
                class="progress-bar-filler"
                style="width: 0%; background: rgb(136, 136, 136);"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
